### ISSUE TITLE
Loss is NaN

### ISSUE THREAD

**Issue by abhisheksoni27** [subpath/Keras_music_gereration#2]
While training, the loss is `NaN` all the time. Logging the code in the earlier steps, I figured that the `combine` variable you are using to combine notes and velocities has a `NaN` in every element.

It becomes NaN after this block:

```py
note_min = np.min(notes)
note_max = np.max(notes)
velocities_min = np.min(velocities)
velocities_max = np.max(velocities)

for i in combine:
    i[0] = 2*(i[0]-((note_min+note_max)/2))/(note_max-note_min)
    i[1] = 2*(i[1]-((velocities_min+velocities_max)/2))/(velocities_max-velocities_min)

```

`combine` new value: 

```
[[-0.3333333333333333, nan],
 [0.3333333333333333, nan],
 [-0.3888888888888889, nan],
 [0.2777777777777778, nan],
 [-0.5, nan],
 [0.16666666666666666, nan],
 [-0.6111111111111112, nan],
 ...,
 ...,
 ...,
]
```

Caused by division by 0. (`velocities_min` is equal to `velocities_max`, so their difference becomes 0.)

**Comment by subpath** [comment]
Hm, you're right, I'm gettings NaN's too. But I'm pretty sure that wasn't a case when I submitted the first time, you can see it in notebook's outputs. Let me investigate this. Cheers!

**Comment by abhisheksoni27** [comment]
Yes, I saw that and was pretty excited to try it on another midi. I'm sure it is just a minor bug but I can't figure it out. While you are at it, can you also explain why and what we are doing here?

```js
for i in combine:
    i[0] = 2*(i[0]-((note_min+note_max)/2))/(note_max-note_min)
    i[1] = 2*(i[1]-((velocities_min+velocities_max)/2))/(velocities_max-velocities_min)

```

**Comment by subpath** [comment]
In order to fit data into the model, I needed to apply scaling, for some reason back then I decided, that it would be a good idea to write my own min-max scaler instead of using ready solution, for example from sklearn.
To be completely honest, that was one of my first repo here, I was just trying something with LSTM first time)  I really need to refactor that code.

**Comment by hbina** [comment]
The problem is that the values of velocities is **always** 67. So min == max and you ended up with division by zero when performing 

>     for i in combine:
>        print(i, note_min, note_max, velocities_min, velocities_max)
>        i[0] = 2 * (i[0] - ((note_min + note_max) / 2)) / (note_max - note_min)
>        i[1] = 2 * (i[1] - ((velocities_min + velocities_max) / 2)) / (velocities_max - velocities_min)

Do you have a plan to fix this? I am sure not sure how to myself

**Comment by subpath** [comment]
Please pull from master one more time. I fixed Nan as you mentioned. 
Midi is a structured format, so it very easy to convert it to a matrix of numbers
